src/main/java/com/strandls/esmodule/services/impl/ElasticSearchServiceImpl.java

```diff
@@ -996,6 +1003,46 @@ private AggregationResponse groupAggregation(String index, AggregationBuilder ag
 					groupMonth.put(entry.getKeyAsString() + "_" + month, monthSumDays.getOrDefault(month, (long) 0));
 				}
 			}
+		} else if (filter.equals(Constants.GROUP_BY_TAXON)) {
+			Terms termsHistogram = response.getAggregations().get("taxon_agg");
+			Map<String, Long> fromMonth = new LinkedHashMap<>();
+			for (Terms.Bucket entry : termsHistogram.getBuckets()) {
+				fromMonth.put(entry.getKeyAsString(), entry.getDocCount());
+			}
+			Map<String, Object> afterKey = Map.of("path", "1");
+			for (int i = 1; i <= 10; i++) {
```

Comment:
what does a hard coded value 10 signify here? also can we rename iterator variable i to something else which clarifies what we are iterating on? (eg taxonObjectIndex, etc)

Reply:
Replaced the for loop with while loop as we don't use the iterator.